src/main/java/WayofTime/bloodmagic/api/registry/AltarRecipeRegistry.java

```diff
@@ -82,17 +88,46 @@ public AltarRecipe(ItemStack input, ItemStack output, EnumAltarTier minTier, int
             this(input, output, minTier, syphon, consumeRate, drainRate, false);
         }
 
+        /**
+         * {@link OreDictionary} version of the above method
+         */
+        public AltarRecipe(String oreDictInput, ItemStack output, EnumAltarTier minTier, int syphon, int consumeRate, int drainRate)
+        {
+            this(OreDictionary.getOres(oreDictInput), output, minTier, syphon, consumeRate, drainRate, false);
```

Comment:
Again, ***do not call this unless you have called `OreDictionary.doesOreNameExist()` first.***

`OreDictionary.getOres()` obtains a list of ItemStacks from the provided entry. If the entry doesn't exist it *creates one and returns an empty list*. That is *bad*. `OreDictionary.doesOreNameExist()` checks if the entry is in the map *without* adding an empty entry.